massa-execution/src/types.rs

```diff
@@ -25,26 +26,49 @@ pub(crate) struct StepHistoryItem {
 }
 
 #[derive(Clone)]
-/// Stateful context, providing an execution context to massa between module executions
+/// Stateful context, providing a context during the execution of a module
 pub(crate) struct ExecutionContext {
+    /// final and active ledger at the current step
     pub ledger_step: SCELedgerStep,
+
+    /// max gas for this execution
     pub max_gas: u64,
+
+    /// coins transferred at the launch of the execution
     pub coins: Amount,
+
+    /// gas price of the execution
     pub gas_price: Amount,
+
+    /// slot at which the execution happens
     pub slot: Slot,
+
+    /// counter of newly created addresses so far during this execution
     pub created_addr_index: u64,
+
+    /// block ID, if one is present at this slot
     pub opt_block_id: Option<BlockId>,
+
+    /// block creator addr, if there is a block at this slot
     pub opt_block_creator_addr: Option<Address>,
+
+    /// call stack, most recent is at the back
     pub call_stack: VecDeque<Address>,
+
+    /// list of addresses created so far during excution
     pub owned_addresses: Set<Address>,
+
+    /// True if it's a read-only context
     pub read_only: bool,
 }
 
+/// an active execution step request
```

Comment:
I'm confused by the "request", Isn't it something else ?

Reply:
clarified